--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -839,3 +839,4 @@
 	sun50i-h616-orangepi-zero2.dtb \
 	sun50i-h618-orangepi-zero3.dtb \
-	sun50i-h616-x96-mate.dtb
+	sun50i-h616-x96-mate.dtb \
+	sun50i-h616-h20-pro-5g.dtb
 dtb-$(CONFIG_MACH_SUN50I) += \
--- /dev/null
+++ b/arch/arm/dts/sun50i-h616-h20-pro-5g.dts
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2024 Armbian
+ */
+
+/dts-v1/;
+
+#include "sun50i-h616.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+
+/ {
+	model = "H20 Pro 5G";
+	compatible = "hechuang,h20-pro-5g\0allwinner,sun50i-h616";
+
+	aliases {
+		serial0 = &uart0;
+		ethernet0 = &emac1;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+&i2c0 {
+	status = "disabled";
+};
+
+&mmc0 {
+	broken-cd;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&mmc2 {
+	bus-width = <8>;
+	non-removable;
+	cap-mmc-hw-reset;
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&emac1 {
+	phy-mode = "rmii";
+	status = "okay";
+};
+
