--- a/arch/arm64/boot/dts/allwinner/Makefile
+++ b/arch/arm64/boot/dts/allwinner/Makefile
@@ -1,5 +1,6 @@
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-amarula-rambutan.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-pine64-plus.dtb
+dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h616-h20-pro-5g.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h616-orangepi-zero.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h616-orangepi-zero2.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h616-orangepi-zero2-w.dtb
+--- /dev/null
++++ b/arch/arm64/boot/dts/allwinner/sun50i-h616-h20-pro-5g.dts
+@@ -0,0 +1,58 @@
++// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
++/*
++ * Copyright (C) 2024 Armbian
++ */
++
++/dts-v1/;
++
++#include "sun50i-h616.dtsi"
++#include <dt-bindings/gpio/gpio.h>
++#include <dt-bindings/interrupt-controller/arm-gic.h>
++
++/ {
++	model = "H20 Pro 5G";
++	compatible = "hechuang,h20-pro-5g\0allwinner,sun50i-h616";
++
++	aliases {
++		serial0 = &uart0;
++		ethernet0 = &gmac1;
++	};
++
++	chosen {
++		stdout-path = "serial0:115200n8";
++	};
++};
++
++&mmc2 {
++	pinctrl-0 = <&sdc2_pins>;
++	pinctrl-1 = <&sdc2_sleep_pins>;
++	pinctrl-names = "default"\0"sleep";
++	bus-width = <8>;
++	cap-mmc-highspeed;
++	cap-cmd23;
++	mmc-cache-ctrl;
++	non-removable;
++	max-frequency = <200000000>;
++	cap-erase;
++	mmc-high-capacity-erase-size;
++	no-sdio;
++	no-sd;
++	status = "okay";
++};
++
++&mmc0 {
++	pinctrl-0 = <&sdc0_pins>;
++	pinctrl-1 = <&sdc0_sleep_pins>;
++	pinctrl-names = "default"\0"sleep";
++	bus-width = <4>;
++	cap-sd-highspeed;
++	cap-wait-while-busy;
++	no-sdio;
++	no-mmc;
++	sd-uhs-sdr50;
++	sd-uhs-ddr50;
++	sd-uhs-sdr104;
++	status = "okay";
++};
++
++&uart0 {
++	pinctrl-names = "default"\0"sleep";
++	pinctrl-0 = <&uart0_pins>;
++	pinctrl-1 = <&uart0_sleep_pins>;
++	status = "okay";
++};
++
++&gmac1 {
++	pinctrl-0 = <&gmac1_pins>;
++	pinctrl-1 = <&gmac1_sleep_pins>;
++	pinctrl-names = "default"\0"sleep";
++	phy-mode = "rmii";
++	tx-delay = <0x07>;
++	rx-delay = <0x1f>;
++	phy-rst;
++	status = "okay";
++};
